driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java

```diff
@@ -108,9 +111,47 @@ public ChangeStreamDocument(@BsonProperty("operationType") final String operatio
         this.txnNumber = txnNumber;
         this.lsid = lsid;
         this.wallTime = wallTime;
+        this.splitEvent = splitEvent;
         this.extraElements = extraElements;
     }
 
+    /**
+     * Creates a new instance
+     *
+     * @param operationTypeString the operation type
+     * @param resumeToken the resume token
+     * @param namespaceDocument the BsonDocument representing the namespace
+     * @param destinationNamespaceDocument the BsonDocument representing the destinatation namespace
+     * @param fullDocument the full document
+     * @param fullDocumentBeforeChange the full document before change
+     * @param documentKey a document containing the _id of the changed document
+     * @param clusterTime the cluster time at which the change occured
+     * @param updateDescription the update description
+     * @param txnNumber the transaction number
+     * @param lsid the identifier for the session associated with the transaction
+     * @param wallTime the wall time of the server at the moment the change occurred
+     * @param extraElements any extra elements that are part of the change stream document but not otherwise mapped to fields
+     *
+     * @since 4.7
+     */
+    @Deprecated
+    public ChangeStreamDocument(@BsonProperty("operationType") final String operationTypeString,
```

Comment:
This is an unfortunate proliferation of constructors, but it seems in line with what was done previously in this class.

Reply:
Should we include a "@deprecated" tag to specify the reason for deprecation?

Reply:
I added the following just now,
```
     * @deprecated Prefer a non-deprecated constructor
```
but when I read it, it seemed too obvious, like "@deprecated use something that isn't deprecated". I think one convention (apart from not including the tag) is to link to the preferred replacement, but here, I do not think we should do this since there have already been N replacements, so it seemed like a link would become outdated.

If you still think this is worth including, I don't feel too strongly, but let me know the wording.